src/main/java/org/elasticsearch/kafka/indexer/jobs/ConsumerStartOption.java

```diff
@@ -54,37 +55,51 @@ public ConsumerStartOption(String property) throws IllegalArgumentException {
 	}
 
 	public static Map<Integer, ConsumerStartOption> fromFile(String configFilePath) throws IllegalArgumentException {
-		Map<Integer, ConsumerStartOption> config = new HashMap<>();
-		if (!StringUtils.isEmpty(configFilePath)) {
-			File configFile = new File(configFilePath);
-		if (configFile.exists()) {
-			try {
-				List<String> lines = Files.readAllLines(configFile.toPath());
-				lines.stream()
-						//filter empty lines and comments (lines starts with '#')
-						.filter(line -> !line.isEmpty() && !line.startsWith("#"))
-						.forEach(line -> {
-							ConsumerStartOption option = new ConsumerStartOption(line);
-							config.put(option.getPartition(), option);
-						});
-			} catch (IOException e) {
-				String message = "Unable to read Consumer start options configuration file from '" +
-						configFile.getPath() + "'";
-				logger.error(message);
-				throw new IllegalArgumentException(message);
-			}
-		} else {
+		final Map<Integer, ConsumerStartOption> config = new HashMap<>();
+		if (StringUtils.isEmpty(configFilePath)) {
+			logger.info("Consumer start options configuration file is not defined. Consumer will use 'RESTART' option by default");
+			return config;
+		}
+		File configFile = new File(configFilePath);
+		if ( !configFile.exists()) {
 			logger.warn("Consumer start options configuration file '"
-					+ configFile.getPath() + "' doesn't exist. Consumer will use 'RESTART' option by default");
+				+ configFile.getPath() + "' doesn't exist. Consumer will use 'RESTART' option by default");
+			return config;
 		}
-		} else {
-			logger.info("Consumer start options configuration file is not defined. Consumer will use 'RESTART' option by default");
+		try {
+			List<String> lines = Files.readAllLines(configFile.toPath());
+			lines.stream()
+					//filter empty lines and comments (lines starts with '#')
+					.filter(line -> !line.isEmpty() && !line.startsWith("#"))
+					.forEach(line -> {
+						ConsumerStartOption option = new ConsumerStartOption(line);
+						config.put(option.getPartition(), option);
+					});
+		} catch (IOException e) {
+			String message = "Unable to read Consumer start options configuration file from '" +
+					configFile.getPath() + "'";
+			logger.error(message);
+			throw new IllegalArgumentException(message);
```

Comment:
You re loosing original exception here. Why throw `IllegalArgumentException`? It can be any kind of IO issues.

Reply:
will leave as is - the thrown IllegalArg exception is eventually thrown out from the ConsumerManager's init() method  and it correctly states that the indexer did not start due to issues with the input configuration options